### DISCUSSION TITLE
Missing Rows in external DB

### DISCUSSION THREAD

**Discussion by bernhard-teacher** [Budibase/budibase#10198] [Help] [answered]
Hi,
i'm using BudiBase with an external PostgreSQL Database.

I'm very confused with one table in my DB:
* Data Sources: Rows missing
* Screen with Table-Block: All rows are displayed
* relationship-Picker: Value is null
* with an custom-Query all rows are displayed
* pgadmin: all rows are displayed

If i export the app and reimport it, all is the same.
If i backup the database and restore it to another DB, i see all the rows in this table!?

I don't know if this issue is caused on BudiBase or on the PostgreSQL-DB...
What infos are stored in the internal CouchDB - can it be an conflict in an document there??? Like this:
https://github.com/Budibase/budibase/discussions/7777#discussioncomment-3732390

Thanks for any suggestions and ideas!!!

**Comment by ConorWebb96** [comment]
Hey @bernhard-teacher,

There isn't a whole lot to go on here, could we possibly get an app export as well as the Postgres schema so we can have a more in-depth look!

Thanks in advance.

**Reply by bernhard-teacher** [reply]
Hi @ConorWebb96 ,
I'd love to, how can I send you the files - not public please.

**Reply by andz-bb** [reply]
hey @bernhard-teacher

You can either send the app export directly to Conor on our [Budibase Discord server](https://discord.gg/ZepTmGbtfF) or by email to the support team at support@budibase.com :)

**Reply by bernhard-teacher** [reply]
Hi @andz-bb 
i sent it to support via Email, but no response yet. Has it arrived you or Conor?
Thx for a short status-update!

**Reply by ConorWebb96** [reply]
Hey @bernhard-teacher,

Sorry, for the long response time! 

The email was archived for some unknown reason and due to that we missed it. I've downloaded and imported the Postgres sql dump and imported the app. I noticed no data was available within the sql dump. I also seen a few error throughout but I'm assuming these are due to the lack of data.

I haven't been able to recreate the issue you are experiencing. I've created about 10 rows for a few different tables and can see all of them.

Is there any additional information you can provide to help us, with finding the error you've described?

Thanks!

**Reply by bernhard-teacher** [reply, marked as answer]
Thanks @ConorWebb96 for your reply.
I have fixed the issue now:
* Backup DB-schema
* drop schema
* restore schema
I don't know, why is it fixed, the restore contains all the settings/data from the previous version - but now it works!

Thanks for your support and happy Weekend :-)

**Comment by bernhard-teacher** [comment]
Hi, i have to reopen this discussion :(
The problem is still there and new strange things...
In the table 'bildungsveranstaltung' are 30 rows. In the BudiBase Screen are only 25 Rows visible:
![2023-06-14 13_26_23-BiVeMa](https://github.com/Budibase/budibase/assets/118993769/412801b6-152d-495b-ac65-ea8f0c4dcd32)

If i sort over the column BVNUMMER more rows are gone..
![2023-06-14 13_27_14-BiVeMa](https://github.com/Budibase/budibase/assets/118993769/3b08626b-71f1-4ff6-b63b-112f729c1c27)

In the Design-Tab there is no filter in the Table block, no Pagination, Scroll limit is 100.
What the hell is going on here????

**Comment by peterhuppertz** [comment]
I might have something similar. I'm on MySQL and have `Table A` connected via `1-to-many relation` to `Table B`. This all worked fine, but now as soon as I add another entry in `Table A` referencing `Table B` some of the rows in `Table B` are gone. Using MySQL Workbench I see they are all still there, but not shown in table view of Budibase nor in the Relationship picker in the form:
<img width="445" alt="image" src="https://github.com/Budibase/budibase/assets/5812004/f4da83a2-57a4-4535-bad9-9b8f916e8a78">
Only Cash and CCC show

<img width="138" alt="image" src="https://github.com/Budibase/budibase/assets/5812004/cd1cf471-2dae-4768-86a8-d0f5f1109871">
All 5 entries are shown in MySQL Workbench

As soon as I delete the last entry from `Table A` all values in `Table B` show up again. However, I don't see any difference to other rows in that table that might create an error condition. I also don't see any related error messages. Any idea what is going on?

**Reply by melohagan** [reply]
Hey @peterhuppertz Would you be able to share your SQL schemas please? https://docs.budibase.com/docs/troubleshooting-sql

I can then try to reproduce. Could you also provide screenshots of your define relationships modals please.

Also how many rows do you have in total across both of your tables?

**Comment by dado-ca** [comment]
Same issue here, on Postgre external database. Two tables with one to many relation ship, department->department detail. There is also a third table, Employees, which has a relationship to the department table. As soon as I create a row in the Employeetable, containing an department id, some of the records in the department table dissapeare. Howewer all the records are visible in pgadmin. Strange.

**Reply by melohagan** [reply]
Hey @dado-ca Are you able to share the SQL schemas? How many rows are in each of the tables?

**Reply by dado-ca** [reply]
I am no allowed, but the same issue occures in other three tables, interval (1)->scheduleInterval (N) , with one to many relationship. In the third table Timeattendance which has a relationship to the interval table, one to many interval (1) -> Timeattendance (N), the same thing happens.

 As soon as I create some records in the TimeAttendance table, containing ID's of the rows in the  interval table, some rows in the interval table dissapeare. 

 Some explanation:  I have created an Time tracking app for our Employees, so the **Interval** table contains list of  working obligations, work breaks  and abscence.... like  regular work (RED), lunch break (RUČ),  sick day (BOL) etc. 

The **IntervalSchedule** table contains time ranges in which intervals are effective, for example ,  from 7:30 - 10:00  the RED interval apply, from 10:00 - 10:30 RUČ (lunch break) interval apply, from 10:30-16:30 RED interval apply, those are defined for days monday to friday.

The **TimeAtendance**  table is  my main table which contains all timeattendance tracking data, employee id, date, interval, begin time, end time, duration.

**I must say, I solved this issue** buy deleting the budibase relationship between the interval table and timeattendance table. Maybe bad database design but i dont see how.

**Reply by melohagan** [reply]
Perhaps you defined the relationships in Budibase back to front? I.e. a Many->One instead of a One->Many ? Glad you solved the issue though!

**Comment by tranx** [comment]
have similar issue with postgres. What is weird is that when I change the order of the rows. The budibase view of the table will return 1,16, or 18 rows... I imagine that if I remove the relationship things are fine, but that defeat the point of budibase if people have to start to know id of rows in other tables to be able to be able to select a record properly... It looks like it is related to order of primary key vs order of display columns / relationship - but why ...
